src/legacy/server/saved_objects/import/validate_references.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObject, SavedObjectsClient } from '../service';
+import { ImportError } from './types';
+
+const ENFORCED_TYPES = ['index-pattern', 'search'];
+
+export async function validateReferences(
+  savedObjects: SavedObject[],
+  savedObjectsClient: SavedObjectsClient
+) {
+  const errors: ImportError[] = [];
+
+  // Grab record ids of the objects existing in Elasticsearch
+  const findResponse = await savedObjectsClient.find({
```

Comment:
`savedObjectsClient.find()` only returns 20 results by default, I think that since we have all the ids we are interested in validating we should use `savedObjectsClient.bulkGet()` instead, and maybe add a way to limit that method so it only returns enough data for use to determine existence...

Alternatively, we could ask for all the results back in a single page, but that feels like a bad idea too...